core/services/keeper/registry_interface.go

```diff
@@ -27,6 +28,10 @@ const (
 
 const ActiveUpkeepIDBatchSize int64 = 10000
 
+type UpkeepGetter interface {
+	GetUpkeep(*bind.CallOpts, *big.Int) (*UpkeepConfig, error)
+}
+
```

Comment:
I'm not sure if we need this interface, ref to comments below

Reply:
See larger discussion below.